lib/elixir/lib/integer.ex

```diff
@@ -60,6 +60,88 @@ defmodule Integer do
   end
 
   @doc """
+  Computes the modulo remainder of an integer division.
+
+  `Integer.mod/2` uses floored division, which means that 
+  the result will always have the sign of the `divisor`.
+
+  Raises an `ArithmeticError` exception if one of the arguments is not an
+  integer, or when the `divisor` is `0`.
+
+  Unlike `rem/2`, `Integer.mod/2` is _not_ allowed in guard tests.
```

Comment:
I don't think we need to state this explicitly, if we did for every function that was not allowed in guards we'd have to add a lot of notes :)